src/oop/object_interactions/medium/q6/Airplane.java

```diff
@@ -0,0 +1,54 @@
+package oop.object_interactions.medium.q6;
+
+public class Airplane {
+    private int maxSeats;
+    private int numPassengers;
+
+    public Airplane(int seats) {
+        maxSeats = seats;
+        numPassengers = 0;
+    }
+
+    public Airplane(int seats, int people) {
+        maxSeats = seats;
+        numPassengers = people;
+    }
+
+    public void transferPassengers(Airplane busyPlane) {
+        int emptySeats = maxSeats - numPassengers;
+        if (emptySeats > 0) {
+            int numToMove = busyPlane.getNumPassengers() - busyPlane.getMaxSeats();
+            if (emptySeats > numToMove) {
+                numPassengers += numToMove;
+                busyPlane.setNumPassengers(busyPlane.getNumPassengers() - numToMove);
+                System.out.println("We moved " + numToMove + " passengers");
+            } else {
+                int numCanMove = numToMove - emptySeats;
```

Comment:
this gives an incorrect number which leads to the strange output of
"The first ride: 16 out of  12 seats are filled"
"The second ride: 2 out of  6 seats are filled"
"The third ride: 2 out of  4 seats are filled"
Admittedly, I also missed this on the first pass and only the main method test alerted me. 

Get rid of numCanMove and change all usages of it to emptySeats